test/e2e/external_auth_complete_cluster.go

```diff
@@ -0,0 +1,242 @@
+			if ver == "" {
+				ver = "4.19" // default MAJOR.MINOR required by API
+			}
+			clusterTpl := applyVersionToTemplate(rawClusterTpl, ver)
+
+			managedRG := framework.SuffixName(*resourceGroup.Name, "-managed", 64)
+			_, err = framework.CreateBicepTemplateAndWait(ctx,
+				tc.GetARMResourcesClientFactoryOrDie(ctx).NewDeploymentsClient(),
+				*resourceGroup.Name, "hcp-cluster",
+				clusterTpl,
+				map[string]interface{}{
+					"nsgName":                  customerNSGName,
+					"vnetName":                 customerVnetName,
+					"subnetName":               customerVnetSubnetName,
+					"clusterName":              customerClusterName,
+					"managedResourceGroupName": managedRG,
+				}, 45*time.Minute)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating node pool")
+			_, err = framework.CreateBicepTemplateAndWait(ctx,
+				tc.GetARMResourcesClientFactoryOrDie(ctx).NewDeploymentsClient(),
+				*resourceGroup.Name, "node-pool",
+				framework.Must(TestArtifactsFS.ReadFile("test-artifacts/generated-test-artifacts/standard-cluster-create/nodepool.json")),
+				map[string]interface{}{
+					"clusterName":  customerClusterName,
+					"nodePoolName": customerNodePoolName,
+				}, 45*time.Minute)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("applying ExternalAuth config via RP (using Entra app from Step 1)")
+
+			// 1) Load Entra creds JSON (local loader)
+			secretPath := strings.TrimSpace(os.Getenv("ENTRA_E2E_SECRET_PATH"))
+			if secretPath == "" {
+				secretPath = "test/e2e/out/entra_app_secret.json"
+			}
+			sf := mustLoadEntraSecretStep2(secretPath)
+
+			// 2) Build issuer/audience/client from Entra
+			issuerURL := fmt.Sprintf("https://login.microsoftonline.com/%s/v2.0", sf.TenantID)
+			externalAuthClientID := sf.ClientID
+			cliAudience := sf.ClientID
+			externalAuthClientSecret := ""
+
+			// 3) Build payload
+			authPayload := map[string]interface{}{
+				"id": externalAuthID,
+				"issuer": map[string]interface{}{
+					"url":       issuerURL,
+					"audiences": []string{cliAudience},
+				},
+				"claim": map[string]interface{}{
+					"mappings": map[string]interface{}{
+						"userName": map[string]interface{}{"claim": usernameClaim},
+						"groups":   map[string]interface{}{"claim": groupsClaim},
+					},
+					"validation_rules": []any{},
+				},
+				"clients": []map[string]interface{}{
+					{
+						"component": map[string]interface{}{
+							"name":      externalAuthComponentName,
+							"namespace": externalAuthComponentNS,
+						},
+						"id":     externalAuthClientID,
+						"secret": externalAuthClientSecret,
+					},
+				},
+			}
+
+			body, err := json.Marshal(authPayload)
+			Expect(err).NotTo(HaveOccurred())
+
+			// 4) POST to RP
+			baseURL := strings.TrimSpace(os.Getenv("RP_BASE_URL"))
+			if baseURL == "" {
+				baseURL = "https://127.0.0.1:8443"
+			}
+			rpEndpoint := fmt.Sprintf("%s/api/aro_hcp/v1alpha1/clusters/%s/external_auth_config/external_auths",
+				baseURL, customerClusterName)
```

Comment:
somewhat surprised our generated client doesn't have these. Double check to see if https://github.com/Azure/ARO-HCP/blob/main/internal/api/v20240610preview/generated/externalauths_client.go#L53 or something near it works.